File: 387. First Unique Character in a String.py

# link - https://leetcode.com/problems/first-unique-character-in-a-string/

#space: O(n)
#time: O(n)

class Solution:
    def firstUniqChar(self, s: str) -> int:
        c = Counter(s)
        for i in range(len(s)):
            if c[s[i]] == 1:
                return i
        return -1
        

File: Leetcode/387. First Unique Character in a String.cpp
/**
 * https://leetcode.com/problems/first-unique-character-in-a-string/
 * Time: O(N)
 * Space: O(1)
 */

class Solution {
public:
    int firstUniqChar(string s) {

        int cnt[26]{};

        for (char c : s)
            ++cnt[c - 'a'];

        for (int i = 0; i < s.size(); ++i)
            if (cnt[s[i] - 'a'] == 1)
                return i;

        return -1;
    }
};